api/src/main/java/org/commonjava/indy/data/StoreValidatorRemote.java

```diff
@@ -0,0 +1,162 @@
+import java.util.concurrent.*;
+
+
+@RemoteStoreValidator
+public class StoreValidatorRemote implements StoreValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StoreValidatorRemote.class);
+    private final CountDownLatch httpRequestsLatch = new CountDownLatch(2);
+    private final ExecutorService executorService = Executors.newFixedThreadPool(2);
+    private static final String REMOTE = "remote";
+
+
+    @Override
+    public ArtifactStoreValidateData validate(ArtifactStore artifactStore)
+        throws InvalidArtifactStoreException, MalformedURLException {
+        // Parse Remote Repository for checking if it is a valid URL
+        try {
+            //#region
+//            LOGGER.info("=> ArtifactStore Type: " + artifactStore.getType().singularEndpointName());
+            if (REMOTE.equalsIgnoreCase(artifactStore.getType().singularEndpointName())
+                || artifactStore instanceof RemoteRepository) {
+                // Cast to Remote Repository
+                RemoteRepository remoteRepository = (RemoteRepository) artifactStore;
+                //Validate URL from remote Repository URL
+                Optional<URL> remoteUrl = Optional.of(new URL(remoteRepository.getUrl()));
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Boolean> httpGetStatus = executorService.submit(new HttpGetTask(new HttpGet(remoteUrl.get().toURI())));
+                Future<Boolean> httpHeadStatus = executorService.submit(new HttpHeadTask(new HttpHead(remoteUrl.get().toURI())));
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                // Check for Sucessfull Validation
+                if (httpGetStatus.get() && httpHeadStatus.get()) {
+                    LOGGER.info("=> Succesfull Validation for Remote Repository: " + remoteUrl.get());
+                    return new ArtifactStoreValidateData
+                        .Builder(true)
+                        .setRepositoryUrl(remoteUrl.get().toExternalForm())
+                        .setStoreType(remoteRepository.getType())
+                        .build();
+                }
+            } else {
+                LOGGER.warn("=> It's not Remote Artifact Store Repository: " + artifactStore.getType().singularEndpointName());
+            }
+            //#endregion
+        } catch (MalformedURLException mue) {
+            LOGGER.error("=> Mailformed URL: ", mue);
+            throw new MalformedURLException("=> Mailformed URL: " + ((RemoteRepository) artifactStore).getUrl());
+        } catch (Exception e) {
+            LOGGER.error(" => Not Valid Remote Repository, \n => Exception: " + e);
+            throw new InvalidArtifactStoreException("=> Not valid remote Repository: " + artifactStore.getType().singularEndpointName(), e);
+        } finally {
+            if (executorService.isShutdown()) {
```

Comment:
We don't need to handle service shutdown here...we want to keep it around for future work.